tasks/managed/push-rpms-to-pulp/push-rpms-to-pulp.yaml

```diff
@@ -0,0 +1,316 @@
+      description: The url to the git repo where the release-service-catalog tasks and stepactions to be used are stored
+    - name: taskGitRevision
+      type: string
+      description: The revision in the taskGitUrl repo to be used
+  results:
+    - description: Produced trusted data artifact
+      name: sourceDataArtifact
+      type: string
+  volumes:
+    - name: workdir
+      emptyDir: {}
+  stepTemplate:
+    env:
+      - name: SNAPSHOT_PATH
+        value: $(params.dataDir)/$(params.SNAPSHOT_PATH)
+      - name: DEFAULT_EXCLUDES
+        value: $(params.DEFAULT_EXCLUDES)
+      - name: DATA_DIR
+        value: $(params.dataDir)
+      - name: PULP_DOMAIN
+        value: $(params.PULP_DOMAIN)
+      - name: PULP_SECRET_NAME
+        value: $(params.PULP_SECRET_NAME)
+      - name: RESULTS_DIR_PATH
+        value: $(params.resultsDirPath)
+      - name: ociStorage
+        value: $(params.ociStorage)
+      - name: ociArtifactExpiresAfter
+        value: $(params.ociArtifactExpiresAfter)
+      - name: trustedArtifactsDebug
+        value: $(params.trustedArtifactsDebug)
+      - name: IMAGES_TXT
+        value: $(params.dataDir)/images.txt
+      - name: FILES_DIR
+        value: $(params.dataDir)/files
+      - name: IMAGE_EXPIRES_AFTER
+        value: $(params.ociArtifactExpiresAfter)
+      - name: ORAS_OPTIONS
+        value: $(params.orasOptions)
+      - name: DEBUG
+        value: $(params.trustedArtifactsDebug)
+    volumeMounts:
+      - name: workdir
+        mountPath: "/var/workdir"
+    workingDir: "/var/workdir"
+  steps:
+    - name: use-trusted-artifact
+      computeResources:
+        limits:
+          memory: 64Mi
+        requests:
+          memory: 64Mi
+          cpu: 30m
+      ref:
+        resolver: "git"
+        params:
+          - name: url
+            value: $(params.taskGitUrl)
+          - name: revision
+            value: $(params.taskGitRevision)
+          - name: pathInRepo
+            value: stepactions/use-trusted-artifact/use-trusted-artifact.yaml
+      params:
+        - name: workDir
+          value: $(params.dataDir)
+        - name: sourceDataArtifact
+          value: $(params.sourceDataArtifact)
+    - name: push-rpms-to-pulp
+      image: quay.io/konflux-ci/release-service-utils:d39d3af868c7136f7e4bffb32d1e625398d6264a
+      computeResources:
+        limits:
+          memory: 256Mi
+        requests:
+          memory: 256Mi
+          cpu: 250m
+      env:
+        - name: CONFIG_FILE_CONTENT
+          valueFrom:
+            secretKeyRef:
+              name: $(params.PULP_SECRET_NAME)
+              key: cli.toml
+      script: |
+        #!/bin/bash
+        set -euo pipefail
+
+        set +x
+        echo "${CONFIG_FILE_CONTENT:?}" > "/tmp/cli.toml"
+        export PULP_CONFIG_FILE="/tmp/cli.toml"
+        set -x
+
+        DOMAIN_EXISTS=$(pulp --config "$PULP_CONFIG_FILE" domain show --name "${PULP_DOMAIN}" \
+          | jq -r ".name // \"\"" || echo "")
```

Comment:
I'm confused why we have the default of `""` *and* echo `""`. Do we need both?

Reply:
in the event pulp throws an error or does not return a json file, then the `|| echo ""` will take precedence.

if you a suggestion in mind, let me know